src/mpid/ch4/src/ch4_coll_impl.h

```diff
@@ -581,6 +660,36 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_Reduce_intra_composition_beta(const void *sen
     goto fn_exit;
 }
 
+#ifdef MPIDI_BUILD_CH4_SHM
+#undef FUNCNAME
+#define FUNCNAME MPIDI_Reduce_intra_composition_gamma
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX int MPIDI_Reduce_intra_composition_gamma(const void *sendbuf,
```

Comment:
Compositions set is a pretty huge and relatively independent thing. I would move it into a separate commit